logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/message/SessionResetSenderTest.kt

```diff
@@ -0,0 +1,142 @@
+            provideClientId,
+            messageSender,
+            testDispatchers
+        )
+
+        given(slowSyncRepository)
+            .getter(slowSyncRepository::slowSyncStatus)
+            .whenInvoked()
+            .thenReturn(completeStateFlow)
+
+    }
+
+    @Test
+    fun givenClientIdProvideAFailure_whenSendingSessionResetMessage_thenReturnFailure() = runTest(testDispatchers.io) {
+
+        given(provideClientId)
+            .suspendFunction(provideClientId::invoke)
+            .whenInvoked()
+            .thenReturn(Either.Left(failure))
+
+        val result = sessionResetSender(CONVERSATION_ID, USER_ID, CLIENT_ID)
+
+        verify(provideClientId)
+            .suspendFunction(provideClientId::invoke)
+            .wasInvoked(exactly = once)
+
+        assertEquals(Either.Left(failure), result)
+
+    }
+
+    @Test
+    fun givenMessageSenderFailure_whenSendingSessionResetMessage_thenReturnFailure() = runTest(testDispatchers.io) {
+
+        given(provideClientId)
+            .suspendFunction(provideClientId::invoke)
+            .whenInvoked()
+            .thenReturn(Either.Right(CLIENT_ID))
+
+        given(messageSender)
+            .suspendFunction(messageSender::sendMessage)
+            .whenInvokedWith(anything(), anything())
+            .thenReturn(Either.Left(failure))
+
+        val result = sessionResetSender(CONVERSATION_ID, USER_ID, CLIENT_ID)
+
+        verify(provideClientId)
+            .suspendFunction(provideClientId::invoke)
+            .wasInvoked(exactly = once)
+
+        verify(messageSender)
+            .suspendFunction(messageSender::sendMessage)
+            .with(anything(), anything())
+            .wasInvoked(exactly = once)
+
+        assertEquals(Either.Left(failure), result)
+    }
+
+    @Test
+    fun givenMessageSenderRanSuccessfully_whenSendingSessionResetMessage_thenReturnSuccess() = runTest(testDispatchers.io) {
+
+        given(provideClientId)
+            .suspendFunction(provideClientId::invoke)
+            .whenInvoked()
+            .thenReturn(Either.Right(CLIENT_ID))
+
+        given(messageSender)
+            .suspendFunction(messageSender::sendMessage)
+            .whenInvokedWith(anything(), anything())
+            .thenReturn(Either.Right(Unit))
+
+        val result = sessionResetSender(CONVERSATION_ID, USER_ID, CLIENT_ID)
+
+        verify(provideClientId)
+            .suspendFunction(provideClientId::invoke)
+            .wasInvoked(exactly = once)
+
+        verify(messageSender)
+            .suspendFunction(messageSender::sendMessage)
+            .with(anything(), anything())
+            .wasInvoked(exactly = once)
+
+        assertEquals(Either.Right(Unit), result)
+    }
+
+    companion object {
+        val SELF_USER_ID = UserId("self-user-id", "domain")
+        val CONVERSATION_ID = ConversationId("conversation-id", "domain")
+        val CLIENT_ID = ClientId("client-id")
+        val USER_ID = UserId("client-id", "domain")
```

Comment:
I think we can reuse the nice stubs we have on `com.wire.kalium.logic.framework` ie: `TestClient` :D